### PR TITLE
[feat] Add command-line option to suppress deprecation warnings

### LINKED ISSUE #471: Add option to suppress deprecation warnings

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #471

### CODE REVIEW

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: unittests/test_cli.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from contextlib import redirect_stdout, redirect_stderr
 from io import StringIO
 
+from reframe.core.exceptions import ReframeDeprecationWarning
```
Comment (ekouts) [inline unittests/test_cli.py]: I think you should move this line after `import unittests.fixtures as fixtures`

File: unittests/test_cli.py [hunk 2]
```diff
@@ -443,6 +459,7 @@ def test_verbosity(self):
         assert 'Traceback' not in stderr
         assert 0 == returncode
 
+
```
Comment (ekouts) [inline unittests/test_cli.py]: Do you want the extra empty line here?

File: reframe/frontend/cli.py [hunk 3]
```diff
@@ -151,6 +152,9 @@ def main():
     run_options.add_argument(
         '--reservation', action='store', metavar='RES',
         help='Use RES for submitting jobs')
+    run_options.add_argument(
+        '--suppress-deprecation', action='store_true',
```
Comment (vkarak) [inline reframe/frontend/cli.py]:
```suggestion
        '--no-deprecation-warnings', action='store_true',
```

File: unittests/test_cli.py [hunk 4]
```diff
@@ -433,6 +434,21 @@ def test_show_env_config_unknown_env(self):
...
+    def test_suppress_deprecation(self):
+        self.action = 'run'
+        self.checkpath = [
+            'unittests/resources/checks_unlisted/deprecated_test.py'
+        ]
+        with pytest.warns(ReframeDeprecationWarning):
+            returncode, stdout, stderr = self._run_reframe()
+
+        self.more_options = ['--suppress-deprecation']
+        with pytest.warns(None) as record:
```
Comment (vkarak) [inline unittests/test_cli.py]:
```suggestion
        with pytest.warns(None) as warnings:
```

File: unittests/test_cli.py [hunk 5]
```diff
@@ -433,6 +434,21 @@ def test_show_env_config_unknown_env(self):
...
+            self._run_reframe()
+
+        assert not any(isinstance(r.message, ReframeDeprecationWarning) for
+                       r in record)
```
Comment (vkarak) [inline unittests/test_cli.py]: Where is the documentation for the return value of the `pytest.warns()`? I had to look into the code, but it's a bit hard to understand how a message can be of an exception type.

```suggestion
        assert not any(isinstance(w.message, ReframeDeprecationWarning)
                       for w in warnings)
```

**Pull request merged**